test/extensions/soulbound.ts

```diff
@@ -106,51 +110,122 @@ describe('RMRKSoulboundNestableExternalEquippableMock', async function () {
   shouldBehaveLikeSoulboundNestable();
 });
 
-describe('RMRKSoulbound exempt', async function () {
-  let token: RMRKSemiSoulboundNestableMock;
+describe('RMRKSoulbound variants', async function () {
   let owner: SignerWithAddress;
   let otherOwner: SignerWithAddress;
 
   beforeEach(async function () {
-    const signers = await ethers.getSigners();
-    owner = signers[0];
-    otherOwner = signers[1];
-    const factory = await ethers.getContractFactory('RMRKSemiSoulboundNestableMock');
-    token = <RMRKSemiSoulboundNestableMock>await factory.deploy('Chunky', 'CHNK');
-    await token.deployed();
+    [owner, otherOwner] = await ethers.getSigners();
   });
 
-  it('can support IRMRKSoulbound', async function () {
-    expect(await token.supportsInterface(IRMRKSoulbound)).to.equal(true);
+  describe('RMRKSoulbound After Blocks', async function () {
+    const blocksToTransfer = 100;
+    let token: RMRKSoulboundAfterBlockMock;
+    let initBlock: number;
+
+    beforeEach(async function () {
+      const factory = await ethers.getContractFactory('RMRKSoulboundAfterBlockMock');
+      initBlock = (await ethers.provider.getBlock('latest')).number;
+      token = <RMRKSoulboundAfterBlockMock>(
+        await factory.deploy('Chunky', 'CHNK', initBlock + blocksToTransfer)
+      );
+      await token.deployed();
+    });
+
+    it('can support IRMRKSoulbound', async function () {
+      expect(await token.supportsInterface(IRMRKSoulbound)).to.equal(true);
+    });
+
+    it('can get last block to transfer', async function () {
+      expect(await token.getLastBlockToTransfer()).to.equal(initBlock + blocksToTransfer);
+    });
+
+    it('can transfer before max block', async function () {
+      await token.mint(owner.address, 1);
+      await token.transferFrom(owner.address, otherOwner.address, 1);
+      expect(await token.ownerOf(1)).to.equal(otherOwner.address);
+    });
+
+    it('cannot transfer after max block', async function () {
+      await token.mint(owner.address, 1);
+      await mine(blocksToTransfer + 1);
+      await expect(
+        token.transferFrom(owner.address, otherOwner.address, 1),
+      ).to.be.revertedWithCustomError(token, 'RMRKCannotTransferSoulbound');
+    });
   });
 
-  it('can transfer if soulbound exempt', async function () {
-    const tokenId = await mintFromMock(token, owner.address);
-    await token.setSoulboundExempt(tokenId);
-
-    await token.transfer(otherOwner.address, tokenId);
-    expect(await token.ownerOf(tokenId)).eql(otherOwner.address);
+  describe('RMRKSoulbound After Transactions', async function () {
```

Comment:
Should this equal the contract name?

```suggestion
  describe('RMRKSoulboundAfterTransactions', async function () {
```